### PR TITLE
SRE-887 Add SNS message attributes

### PR DESCRIPTION
**Pull request by andrba**
If multiple subscribers are subscribed to the SNS topic, all of them receive all messages, regardless of whether they need those messages.

Adding message attributes to SNS messages enables defining Subscription Filter policies that help routing the messages to relevant subscribers

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -16,7 +16,6 @@ repos:
       - id: check-ast
       - id: check-json
       - id: check-merge-conflict
-      - id: check-yaml
```
Comment (andrba) [inline .pre-commit-config.yaml:19]: The linter fails on parsing the cloudformation template syntax, e.g. a `!Sub` function

File: scan_test.py [hunk 2]
```diff
@@ -246,6 +246,10 @@ def test_sns_start_scan(self):
             "TargetArn": sns_arn,
             "Message": json.dumps({"default": json.dumps(message)}),
             "MessageStructure": "json",
+            "MessageAttributes": {
+                "bucket": {"DataType": "String", "StringValue": self.s3_bucket_name},
+                "key": {"DataType": "String", "StringValue": self.s3_key_name},
```
Comment (msxavi) [inline scan_test.py:251]: Is it possible to mock `bucket_name` and `key_name` and use literals here?

File: scan_test.py [hunk 3]
```diff
@@ -383,6 +387,8 @@ def test_sns_scan_results(self):
             "MessageAttributes": {
                 "av-status": {"DataType": "String", "StringValue": scan_result},
                 "av-signature": {"DataType": "String", "StringValue": scan_signature},
+                "bucket": {"DataType": "String", "StringValue": self.s3_bucket_name},
+                "key": {"DataType": "String", "StringValue": self.s3_key_name},
```
Comment (msxavi) [inline scan_test.py:391]: Same as above.

File: scan_test.py (hunk 2 above)
Comment (andrba) [reply scan_test.py:251]: I didn't want to go against the code style of that repo. `bucket_name` and `key_name` are test variables defined in the setUp method before each test https://github.com/jobready/bucket-antivirus-function/blob/master/scan_test.py#L43-L44 and used ubiquitously throughout the tests. There is no possibility of the application code mutating them.

File: scan_test.py (hunk 3 above)
Comment (andrba) [reply scan_test.py:391]: replied in https://github.com/jobready/bucket-antivirus-function/pull/2/files#r386697550

**Review by msxavi — APPROVED** [review]
👍

**Pull request merged**